features/step_definitions/common_steps.rb

```diff
@@ -336,6 +336,24 @@
   end
 end
 
+Then /^I see an entry in the '(.*)' table with:$/ do |table_heading, table|
+  result_table_rows = get_table_rows_by_caption(table_heading)
+  expected_row = table.rows[0]
+  for row in result_table_rows
+    result_columns = row.all('td')
+    match = true
+    for value, index in expected_row.each_with_index
```

Comment:
The nested loops and overwriting the "match" hurt my head a bit, but it's doing the right thing and I can't immediately suggest a neater way to do it...

Maybe a couple of comments would help, like "checking each field of a row", "reset if a row doesn't match", etc.